### ISSUE TITLE
[typescript-estree]: await in non-async function parses without error

### ISSUE THREAD

**Issue by kaicataldo** [typescript-eslint/typescript-eslint#987]
**What code were you trying to parse?**

```ts
function foo() {
  const bar = await baz();
}
```
[AST Explorer link](https://astexplorer.net/#/gist/f6ce54ad401520e25183f145bf17ee46/latest)

**What did you expect to happen?**
Both `espree` and `babel-eslint` throw a parsing error when using `await` inside a non-`async` function. Should `typescript-estree` also throw an error here? It looks like an error is highlighted in the [TypeScript Playground](http://www.typescriptlang.org/play/index.html#code/GYVwdgxgLglg9mABMOcAUBKRBvAUIxCBAZykQCMBDAJ0QF5FKB3SmMqgL0wG5cBfIA).

**What actually happened?**
Parses without error.

**Versions**

    "@typescript-eslint/eslint-plugin": "^2.3.0",
    "@typescript-eslint/parser": "^2.3.0",
    "eslint": "^6.4.0",
    "typescript": "^3.6.3"

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `2.3.0` |
| `TypeScript`                           | `3.6.3` |
| `node`                                 | `10.16.3` |
| `npm`                                  | `6.11.3` |

**Comment by kaicataldo** [comment]
I'm seeing this with another error, so I'm now wondering if this is intentional (and is intended to be to caught by the TS compiler).

```ts
const foo = 1;
const foo = 2;
```

[AST Explorer Link](https://astexplorer.net/#/gist/aa0551524ccea84f43d74942e8ed37bd/latest)

Espree reports the following parser error: `Identifier 'foo' has already been declared`

`@typescript-eslint/typescript-estree` doesn't throw a parse error.
